pkg/util/winutil/eventlog/publishermetadatacache/publishermetadatacache.go

```diff
@@ -0,0 +1,104 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+//go:build windows
+
+// Package publishermetadatacache provides a cache for Windows Event Log publisher metadata handles
+package publishermetadatacache
+
+import (
+	"time"
+
+	"github.com/patrickmn/go-cache"
+
+	evtapi "github.com/DataDog/datadog-agent/pkg/util/winutil/eventlog/api"
+	"golang.org/x/sys/windows"
+)
+
+// InvalidHandle represents an invalid EventPublisherMetadataHandle
+const InvalidHandle = evtapi.EventPublisherMetadataHandle(0)
+
+// PublisherMetadataCache implements the Component interface
+type PublisherMetadataCache struct {
+	cache  *cache.Cache
+	evtapi evtapi.API
+}
+
+// New creates a new publishermetadatacache
+func New(api evtapi.API) *PublisherMetadataCache {
+	// Create cache with 5 minute expiration and handle our own cleanup.
+	// Only using expiration for invalid handles to retry creating the handle once it expires.
+	// Ignore expiration for valid handles.
+	c := cache.New(5*time.Minute, 0)
```

Comment:
(see other comment about FormatMessage returning an error, first)

Since we're only expiring error values, and we're currently not storing (but want to) store the actual error value, I wonder if the logic could be made simpler by keeping a separate cache for handle values and error values.